internal/validation/validation.go

```diff
@@ -32,7 +32,7 @@ const (
 	// NOTE(DongjinS): regular expression is referenced unreserved characters
 	// (https://datatracker.ietf.org/doc/html/rfc3986#section-2.3)
 	// and copied from https://gist.github.com/dpk/4757681
-	slugRegexString                   = `^[a-z0-9\-._~]+$`
+	slugRegexString                   = `^[a-zA-Z0-9\-._~]+$`
```

Comment:
The slug regular expression appears to be an expression of URL slug best practices and we use slug regex in the name of the project.
https://rockcontent.com/blog/what-is-url-slug/

How about defining a separate regex for the document key?